zcash_client_sqlite/src/wallet.rs

```diff
@@ -997,12 +998,13 @@ fn parse_block_metadata<P: consensus::Parameters>(
         block_height,
         block_hash,
         Some(sapling_tree_size),
-        if params
+        #[cfg(feature = "orchard")]
+        if _params
```

Comment:
Per my comment above, if we're still going to have the database tables for Orchard in the schema, then for values that don't depend on `orchard` crate types and aren't wallet-specific (like the Orchard tree size for a given block), we probably don't want to feature flag this off.

The main reason for doing so would be if the presence of an Orchard tree size is used as some kind of indicator or switch for later logic. Do we need to use it as such?

Reply:
I don't want to use the presence or absence of data to be used as a logic switch, for sure. However, as I stated above, I think that we should try to fully segregate protocol-specific functionality so that as much `orchard` stuff as possible is removed from the API when the `orchard` flag is disabled; a smaller API surface is going to be less error-prone.